codex-rs/exec/src/event_processor.rs

```diff
@@ -186,18 +194,38 @@ impl EventProcessor {
             EventMsg::TokenCount(TokenUsage { total_tokens, .. }) => {
                 ts_println!(self, "tokens used: {total_tokens}");
             }
-            EventMsg::AgentMessageDelta(AgentMessageDeltaEvent { delta: _ }) => {
-                // TODO: think how we want to support this in the CLI
+            EventMsg::AgentMessageDelta(AgentMessageDeltaEvent { delta }) => {
+                if !self.answer_started {
+                    ts_println!(self, "{}\n", "codex".style(self.italic).style(self.magenta),);
+                    self.answer_started = true;
+                }
+                print!("{delta}");
+                use std::io::Write;
+                if let Err(e) = std::io::stdout().flush() {
+                    eprintln!("Failed to flush stdout: {e}");
+                }
             }
-            EventMsg::AgentReasoningDelta(AgentReasoningDeltaEvent { delta: _ }) => {
-                // TODO: think how we want to support this in the CLI
+            EventMsg::AgentReasoningDelta(AgentReasoningDeltaEvent { delta }) => {
+                if !self.show_agent_reasoning {
+                    return;
+                }
+                if !self.reasoning_started {
+                    ts_println!(
+                        self,
+                        "{}\n",
+                        "thinking".style(self.italic).style(self.magenta),
+                    );
+                    self.reasoning_started = true;
+                }
+                print!("{delta}");
+                use std::io::Write;
+                if let Err(e) = std::io::stdout().flush() {
```

Comment:
I guess flushing on every `print!` is the right thing to do...

Reply:
Same as above about using `expect()`, though you will have to suppress the clippy warning in both cases.